src/game.cpp

```diff
@@ -3758,8 +3758,9 @@ void Game::handlePointingAtNode(const PointedThing &pointed, const ItemDefinitio
 	}
 
 	if (runData.nodig_delay_timer <= 0.0 && isLeftPressed()
-			&& client->checkPrivilege("interact")) {
-		handleDigging(pointed, nodepos, playeritem_toolcap, dtime);
+			&& client->checkPrivilege("interact")
+			&& handleDigging(pointed, nodepos, playeritem_toolcap, dtime)) {
+		meta = NULL;  // If node is removed, pointer to metadata no longer valid.
```

Comment:
instead of resetting and adding bool, just re-call meta = map.getNodeMetadata(nodepos) another time to have fresh meta